### PR TITLE
12 add translations

### PR DESCRIPTION
**Pull request by mhhd2020**
- [x] Add [Flask-Babel](https://python-babel.github.io/flask-babel/) to the project and mark all strings for translation
- [x] Provide German translations
- [x] Use everything still applicable from the Portuguese translations for the old version of the Sketch Map Tool
- [x] link to crowdin in `CONTRIBUTING.md`
- [x] Document Error Translation
- [x] tests
- [x] ruff
- [x] vcr

Closes https://github.com/GIScience/sketch-map-tool/issues/367

### CODE REVIEW

File: client-src/base/languageCookieConsent.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+function getConsentToStoreLanguageSelection(lang) {
+    if (document.cookie === "languageSet=true" || confirm(
+        "If you set a language, cookies are used to store your selection and to apply it on all sub-pages. "
+        + "This data is used only to retrieve your selected language and whether you allowed these cookies.\n\n"
+        + "Wenn Sie eine Sprache wählen, werden Cookies verwendet, damit sie auf allen "
+        + "Unterseiten angewandt wird. Die Cookies werden nur verwendet, um die gewählte Sprache abzufragen und "
+        + "abzufragen, ob Sie der Verwendung dieser Cookies zur Sprachauswahl zugestimmt haben.\n\n"
+        + "Se você definir um idioma, os cookies serão usados para armazenar sua seleção e aplicá-la em todas as "
+        + "subpáginas. Esses dados são usados somente para recuperar o idioma selecionado e para verificar se você "
+        + "permitiu esses cookies.\n\n",
```
Comment (ElJocho) [inline client-src/base/languageCookieConsent.js:15]: Could the be localized?

File: sketch_map_tool/templates/base.html [hunk 2]
```diff
@@ -27,23 +29,33 @@
...
...2.342 0 0 0-.399-2.741c-.808.187-1.681.301-2.591.332zM4.51 8.5c.035.987.176 1.914.399 2.741A13.612 13.612 0 0 1 7.5 10.91V8.5H4.51zm3.99 0v2.409c.91.03 1.783.145 2.591.332.223-.827.364-1.754.4-2.741H8.5zm-3.282 3.696c.12.312.252.604.395.872.552 1.035 1.218 1.65 1.887 1.855V11.91c-.81.03-1.577.13-2.282.287zm.11 2.276a6.696 6.696 0 0 1-.598-.933 8.853 8.853 0 0 1-.481-1.079 8.38 8.38 0 0 0-1.198.49 7.01 7.01 0 0 0 2.276 1.522zm-1.383-2.964A13.36 13.36 0 0 1 3.508 8.5h-2.49a6.963 6.963 0 0 0 1.362 3.675c.47-.258.995-.482 1.565-.667zm6.728 2.964a7.009 7.009 0 0 0 2.275-1.521 8.376 8.376 0 0 0-1.197-.49 8.853 8.853 0 0 1-.481 1.078 6.688 6.688 0 0 1-.597.933zM8.5 11.909v3.014c.67-.204 1.335-.82 1.887-1.855.143-.268.276-.56.395-.872A12.63 12.63 0 0 0 8.5 11.91zm3.555-.401c.57.185 1.095.409 1.565.667A6.963 6.963 0 0 0 14.982 8.5h-2.49a13.36 13.36 0 0 1-.437 3.008zM14.982 7.5a6.963 6.963 0 0 0-1.362-3.675c-.47.258-.995.482-1.565.667.248.92.4 1.938.437 3.008h2.49zM11.27 2.461c.177.334.339.694.482 1.078a8.368 8.368 0 0 0 1.196-.49 7.01 7.01 0 0 0-2.275-1.52c.218.283.418.597.597.932zm-.488 1.343a7.765 7.765 0 0 0-.395-.872C9.835 1.897 9.17 1.282 8.5 1.077V4.09c.81-.03 1.577-.13 2.282-.287z"/>
```
Comment (ElJocho) [inline sketch_map_tool/templates/base.html:38]: move svg definition to file

**Comment by ElJocho** [comment]
@mhhd2020 nice job so far :). The topic of i18n has now resurfaced and we are glad to have your contribution!

**Comment by ElJocho** [comment]
-> to see the translation the commands in the add-translation docs need to be run -> this should be visible in the dev setup docs

**Comment by mhhd2020** [comment]
> @mhhd2020 nice job so far :). The topic of i18n has now resurfaced and we are glad to have your contribution!

👍 Nice to see that the Sketch Map Tool is still flourishing and continues to be improved!

**Comment by matthiasschaub** [comment]
> > @mhhd2020 nice job so far :). The topic of i18n has now resurfaced and we are glad to have your contribution!
> 
> 👍 Nice to see that the Sketch Map Tool is still flourishing and continues to be improved!

Great to hear from you. Yeah, the Sketch Map Tool is in active development :)

**Review by matthiasschaub — APPROVED** [review]

**Pull request merged**